app/models/submission.rb

```diff
@@ -122,6 +122,22 @@ def survey_id
     survey&.id
   end
 
+  def next
+    consultation.submissions.find_by("id > ?", id)
+  end
+
+  def prev
+    consultation.submissions.find_by("id < ?", id)
+  end
+
+  def first
+    consultation.submissions.first
+  end
+
+  def last
+    consultation.submissions.last
```

Comment:
as above - making the order explicit and only using active submissions

```suggestion
    consultation.submissions.active.order(:created_at).last
```